src/main/java/games/strategy/engine/ClientContext.java

```diff
@@ -46,7 +47,7 @@
   private static final ClientContext instance = new ClientContext();
 
   private final GameEnginePropertyReader gameEnginePropertyReader = new GameEnginePropertyReader();
-  private final MapDownloadController mapDownloadController = new MapDownloadController();
+  private final MapDownloadController mapDownloadController = new MapDownloadController(new GameRunner());
```

Comment:
This may indicate that a reference to the `MapDownloadController` should be acquired from `GameRunner` instead of `ClientContext` (i.e. add a `GameRunner::getMapDownloadController()` method).  There should only be a few (ideally one) place where a `GameRunner` instance is created.  And in this case, using a fresh `GameRunner` will mean the Download Maps window displayed by `MapDownloadController` is not associated with the main frame window for the "real" `GameRunner`.

Another way to think about this is that each `GameRunner` may need its own `ClientContext`. 🤔   But that may be going too far.  I think the purpose of `ClientContext` is to provide true singleton instances that shouldn't be specific to any one `GameRunner`.

Reply:
Yes, that was just a dirty hack to avoid the compiler error, couldn't find a better way to pass Amy instance without changing this class completely.